test-integration/liquidate-loan/1_liquidate_by_collateral.js

```diff
@@ -0,0 +1,141 @@
+  const loanTermsRequestInfo = {
+    borrower,
+    recipient,
+    requestNonce: nonces.newNonce(borrower),
+    amount: amountWei.toFixed(0),
+    duration: durationInDays * ONE_DAY,
+    requestTime: currentTimestamp,
+    caller: loansInstance.address,
+  };
+  const loanResponseInfoTemplate = {
+    responseTime: currentTimestamp - 10,
+    interestRate: 4000,
+    collateralRatio: 6000,
+    maxLoanAmount: maxAmountWei.toFixed(0),
+  };
+  const loanTermsRequest = createLoanTermsRequest(loanTermsRequestInfo);
+  const signedResponses = await createMultipleSignedLoanTermsResponses(
+    web3,
+    loanTermsRequest,
+    signers,
+    loanResponseInfoTemplate,
+    nonces,
+  );
+
+  const setLoanTermsResult = await loansInstance.setLoanTerms(
+    loanTermsRequest.loanTermsRequest,
+    signedResponses,
+    borrowerTxConfigWithValue
+  );
+
+  const termsExpiryTime = await settingsInstance.termsExpiryTime();
+  const expiryTermsExpected = await timer.getCurrentTimestampAndSum(termsExpiryTime);
+  const loanIDs = await loansInstance.getBorrowerLoans(borrower);
+  const lastLoanID = loanIDs[loanIDs.length - 1];
+  loans
+    .loanTermsSet(setLoanTermsResult)
+    .emitted(
+      lastLoanID,
+      borrowerTxConfigWithValue.from,
+      recipient,
+      loanResponseInfoTemplate.interestRate,
+      loanResponseInfoTemplate.collateralRatio,
+      loanResponseInfoTemplate.maxLoanAmount,
+      loanTermsRequestInfo.duration,
+      expiryTermsExpected,
+    );
+
+  console.log(`Advancing time to take out loan (current: ${(await timer.getCurrentDate())})...`);
+  const nextTimestamp = await timer.getCurrentTimestampAndSum(minutesToSeconds(2));
+  await timer.advanceBlockAtTime(nextTimestamp);
+  
+  // Take out a loan.
+  console.log(`Taking out loan id ${lastLoanID}...`);
+  const takeOutLoanResult = await loansInstance.takeOutLoan(lastLoanID, amountWei, borrowerTxConfig);
+  loans
+    .loanTakenOut(takeOutLoanResult)
+    .emitted(lastLoanID, borrowerTxConfig.from, amountWei);
+
+  // Set a lower price for Token/ETH.
+  console.log(`Settings final (lower) oracle price: 1 ${tokenName} = ${finalOraclePrice.toFixed(0)} WEI = ${toUnits(finalOraclePrice, 18)} ETHER`);
+  await chainlinkOracle.setLatestAnswer(finalOraclePrice);
+
+  // Get liquidation status.
+  const loanInfo = await loansInstance.loans(lastLoanID);
+  
+  console.log(`Liquidating loan id ${lastLoanID}...`);
+  const initialLiquidatorTokenBalance = await token.balanceOf(liquidatorTxConfig.from);
+  const initialTotalCollateral = await loansInstance.totalCollateral();
+  const liquidateLoanResult = await loansInstance.liquidateLoan(lastLoanID, liquidatorTxConfig);
+  const liquidateEthPrice = await settingsInstance.liquidateEthPrice();
+  const loanPrinter = new LoanInfoPrinter(web3, loanInfo, { tokenName, decimals });
+  const tokensPaymentIn = loanPrinter.getTotalTokensPaymentInLiquidation(finalOraclePrice, liquidateEthPrice);
+  loans
+    .loanLiquidated(liquidateLoanResult)
```

Comment:
Still trying to understand whether any (local) DAI/USDC is sent to the liquidation process in exchange for the collateral?